src/stacked_borrows.rs

```diff
@@ -467,6 +473,72 @@ impl<'tcx> Stack {
 
         Ok(())
     }
+
+    /// Report a descriptive error when `new` could not be granted from `derived_from`.
+    fn grant_error(
+        &self,
+        derived_from: SbTag,
+        new: Item,
+        alloc_id: AllocId,
+        alloc_range: AllocRange,
+        error_offset: Size,
+    ) -> InterpError<'static> {
+        let action = format!(
+            "trying to reborrow {:?}[{:#x}] with {:?} permission via tag {:?}",
```

Comment:
"via tag" sounds odd to me, I think something about "derived from tag" or "from parent tag" or so is more evocative.

Reply:
I'm concerned that both "derived from tag" and "from parent tag" are both too suggestive that only the permission is from the tag.

I went back and forth with some other people and came up with other wordings which are very similar to your original ones :joy: